peppy/project.py

```diff
@@ -99,12 +104,13 @@ class Project(PathExAttMap):
 
     def __init__(
         self,
-        cfg=None,
-        amendments=None,
-        sample_table_index=None,
-        subsample_table_index=None,
-        defer_samples_creation=False,
-        project_dict=None,
+        cfg: str = None,
+        amendments: Union[str | Iterable[str]] = None,
+        sample_table_index: Union[str | Iterable[str]] = None,
+        subsample_table_index: Union[str | Iterable[str]] = None,
+        defer_samples_creation: bool = False,
+        project_dict: dict = None,
```

Comment:
Is it possible have 2 different constructors in Python? Just thinking about this...

or maybe using a helper function to create a Project object from dict, instead of appending these onto the end of the constructor signature ? what do you think?

Reply:
In the past I used alternative constructor approach in Python with use of classmethods, something like this -> https://code-maven.com/slides/python/class-methods-alternative-constructor. I think it's good idea.

Reply:
We can initialize project from dict (dict with all object variables) by calling emty class and than running function from dict.
Now:
`peppy.Project(project_dict="...")
Can be changed to:
`peppy.Project().from_dict(pep_dictionary="...")

Not sure about 2 different constructors...

And in this PR I have added `pd_object` argument

Reply:
What do you think about that, should we change it?

Reply:
I think I would do it that way.  Keep the constructor clean. SO you might also have a `from_pandas` analogous to `from_dict`, or something, then.

Reply:
Ok, I had problems with `@classmethod` , as I think we can't use it there.
Solution: I wrote simple class method that retrieves class. So now it will look like that: 
1) To init class from dict:
`peppy.Project().from_dict(pep_dictionary=___)`
2) To init class from pd object:
`peppy.Project().from_pandas(pandas_df=___)`